demos/google_api_diagnosis.py

```
@@ -0,0 +1,182 @@
+    # Mock the LLM manager to count calls
+    call_count = 0
+    original_generate = None
+    
+    from mad_spark_alt.core.llm_provider import llm_manager
+    from mad_spark_alt.agents import LLMQuestioningAgent
+    
+    # Store original method
+    original_generate = llm_manager.generate
+    
+    # Create a wrapper that counts calls
+    async def counting_generate(*args, **kwargs):
```

Ruff (0.11.9)

Missing return type annotation for private function `counting_generate`

(ANN202)